server/src/main/java/org/apache/iotdb/db/engine/StorageEngineV2.java

```diff
@@ -555,6 +560,62 @@ public void forceCloseAllProcessor() throws TsFileProcessorException {
     }
   }
 
+  public void closeStorageGroupProcessor(String storageGroupPath, boolean isSeq, boolean isSync) {
+    for (DataRegion dataRegion : dataRegionMap.values()) {
+      if (dataRegion.getLogicalStorageGroupName().equals(storageGroupPath)) {
+        for (TsFileProcessor tsFileProcessor : dataRegion.getWorkSequenceTsFileProcessors()) {
+          dataRegion.syncCloseOneTsFileProcessor(isSeq, tsFileProcessor);
+        }
+      }
+    }
+  }
+
+  public TSStatus operatorFlush(TFlushReq req) throws StorageGroupNotSetException {
+    if (req.storageGroups == null) {
+      StorageEngineV2.getInstance().syncCloseAllProcessor();
+      WALManager.getInstance().deleteOutdatedWALFiles();
+    } else {
+      for (String storageGroup : req.storageGroups) {
+        if (req.isSeq == null) {
+          StorageEngineV2.getInstance().closeStorageGroupProcessor(storageGroup, true, req.isSync);
+          StorageEngineV2.getInstance().closeStorageGroupProcessor(storageGroup, false, req.isSync);
+        } else {
+          StorageEngineV2.getInstance()
+              .closeStorageGroupProcessor(
+                  storageGroup, Boolean.parseBoolean(req.isSeq), req.isSync);
+        }
+      }
+    }
+
+    if (req.storageGroups != null) {
+      List<PartialPath> noExistSg =
+          checkStorageGroupExist(PartialPath.fromStringList(req.storageGroups));
+      if (!noExistSg.isEmpty()) {
+        StringBuilder sb = new StringBuilder();
+        noExistSg.forEach(storageGroup -> sb.append(storageGroup.getFullPath()).append(","));
+        throw new StorageGroupNotSetException(sb.subSequence(0, sb.length() - 1).toString(), true);
+      }
+    }
+    return RpcUtils.getStatus(TSStatusCode.SUCCESS_STATUS);
+  }
+
+  /**
+   * @param storageGroups the storage groups to check
+   * @return List of PartialPath the storage groups that not exist
+   */
+  public static List<PartialPath> checkStorageGroupExist(List<PartialPath> storageGroups) {
+    List<PartialPath> noExistSg = new ArrayList<>();
+    if (storageGroups == null) {
+      return noExistSg;
+    }
+    for (PartialPath storageGroup : storageGroups) {
+      if (!LocalSchemaProcessor.getInstance().isStorageGroup(storageGroup)) {
+        noExistSg.add(storageGroup);
+      }
+    }
+    return noExistSg;
+  }
```

Comment:
move this check to the confignode, only send existed storage group to datanode

Reply:
fixed~